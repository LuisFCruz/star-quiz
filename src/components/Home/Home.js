import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { withStyles, Button } from '@material-ui/core';

const styles = theme => ({
  container: {
    background: '#000',
    color: '#fff',
    padding: '10px'
  },
  quizInfo: {
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '1000px',
    margin: '0 auto',
    [theme.breakpoints.up('sm')]: {
      flexDirection: 'row'
    }
  },
  title: {
    textAlign: 'center'
  },
  description: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    flex: '1',
    padding: '50px',
    textAlign: 'center'
  },
  button: {
    background: '#FFAB00',
    color: '#000',
    fontWeight: '700',
    width: '90%',
    '&:hover': {
      background: '#FFAB00',
      color: '#000'
    }
  },
  imageContent: {
    [theme.breakpoints.up('sm')]: {
      maxWidth: '50%'
    }
  },
  media: {
    maxWidth: '450px',
    width: '100%',
    margin: '0 auto'
  }
});

const MyLink = props => <Link to='/quiz' {...props} />;

export class Home extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired
  };

  render() {
    const { classes } = this.props;
    return (
      <div className={classes.container}>
        <h1 className={classes.title}>Star Quiz</h1>
        <div className={classes.quizInfo}>
          <div className={classes.imageContent}>
            <img
              className={classes.media}
              src='/assets/images/darthvader.png'
              alt='Imagem: Dath Vader'
            />
          </div>
          <div className={classes.description}>
            <p>Você conhece os personagens de Star Wars?</p>
            <p>Sim? Então mostre-nos!</p>
            <p>
              Com esse quiz você terá oportunidade de identificar os principais
              personagens de Starwars, marcar pontos e se tornar um expert nesta
              série de filmes maravilhosa!
            </p>
            <Button
              component={MyLink}
              title='jogar'
              aria-label='jogar'
              className={classes.button}
            >
              {' '}
              JOGAR!
            </Button>
          </div>
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(Home);
