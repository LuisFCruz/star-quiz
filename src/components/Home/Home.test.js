import { shallow } from 'enzyme';
import React from 'react';

import { Home } from './Home';

describe('<Home />', () => {
  const props = { classes: {} };
  const wrapper = shallow(<Home {...props} />);

  test('Smoke test', () => {
    expect(Home).toBeDefined();
    expect(wrapper.instance()).toBeInstanceOf(Home);
  });

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
