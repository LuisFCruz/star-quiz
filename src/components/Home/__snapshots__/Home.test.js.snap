// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Home)
    classes={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "button": "Home-button-5",
        "container": "Home-container-1",
        "description": "Home-description-4",
        "imageContent": "Home-imageContent-6",
        "media": "Home-media-7",
        "quizInfo": "Home-quizInfo-2",
        "title": "Home-title-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "button": "Home-button-5",
          "container": "Home-container-1",
          "description": "Home-description-4",
          "imageContent": "Home-imageContent-6",
          "media": "Home-media-7",
          "quizInfo": "Home-quizInfo-2",
          "title": "Home-title-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
